### PR TITLE
libs/libc/wqueue/work_usrthread.c: Correct time calculation in work_p…

### PR DESCRIPTION
**Pull request by jlaitine**
…rocess

This fixes busylooping in work_usrthread, due to incorrect time spec given to sem_timedwait

_SEM_TIMEDWAIT works on absolute time stamps, using CLOCK_REALTIME

Signed-off-by: Jukka Laitinen <jukkax@ssrc.tii.ae>

### CODE REVIEW

File: libs/libc/wqueue/work_usrthread.c [hunk 1]
```diff
@@ -193,15 +193,18 @@ static void work_process(FAR struct usr_wqueue_s *wqueue)
...
-      sec          = next / 1000000;
-      rqtp.tv_sec  = sec;
-      rqtp.tv_nsec = (next - (sec * 1000000)) * 1000;
+      clock_gettime(CLOCK_REALTIME, &rqtp);
+      nsec = 1000 * TICK2USEC(next);
+      sec = nsec / (1000 * 1000 * 1000);
+      rqtp.tv_sec  += sec;
+      rqtp.tv_nsec += (uint32_t)(nsec - sec * 1000 * 1000 * 1000);
 
       _SEM_TIMEDWAIT(&wqueue->wake, &rqtp);
```
Comment (jpaali) [inline libs/libc/wqueue/work_usrthread.c:209]: Implementation of sem_timedwait() has been changed in 33ec242caf3, which I think causes the issue.

Comment (jlaitine) [reply libs/libc/wqueue/work_usrthread.c:209]: Thanks for pointing this out, that's right

**Comment by jpaali** [comment]
Looks fine for me.

**Review by jpaali — APPROVED** [review]

**Comment by jlaitine** [comment]
Updated to match the commits accepted in upstream

**Pull request merged**